### PR TITLE
Convert autosnapshotting to usage tracking to handle both snapshots and usage events.

### PR DESCRIPTION
**Pull request by polina-c**
Add usage tracking events to address
https://dart-review.googlesource.com/c/sdk/+/300862/comment/d7d5036a_21d6a6c7/

https://user-images.githubusercontent.com/12115586/236654615-aa436ded-1b68-4c6c-86ea-2491357b777f.mov

### CODE REVIEW

File: doc/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+https://user-images.githubusercontent.com/12115586/234953319-6f864d25-9854-4126-b4d6-8e114b9045ff.mov
+
+## Configure usage tracking
+
+Use the function `trackMemoryUsage` to configure usage events and auto-snapshotting.
+
+See [example](../more_examples/autosnapshotting/).
+
+We
```
Comment (bkonyi) [inline doc/USAGE.md:23]: Extra newline?

File: doc/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Analyse snapshots
+
+TODO (polina-c): add link to DevTools help section.
+
+## Auto-snapshotting limitations
+
+### By platform
+
+USage tracking does not work for web platform.
```
Comment (bkonyi) [inline doc/USAGE.md:35]: Usage

File: lib/src/usage_tracking/model.dart [hunk 3]
```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
```
Comment (bkonyi) [inline lib/src/usage_tracking/model.dart:1]: Ditto

File: more_examples/autosnapshotting/lib/main.dart [hunk 4]
```diff
@@ -8,14 +8,20 @@ import 'package:flutter/material.dart';
 import 'package:intl/intl.dart';
 import 'package:leak_tracker/leak_tracker.dart';
 
+late String _snapshotDirectory;
+
+extension SizeConversion on int {
+  int mbToBytes() => this * 1024 * 1024;
+  int bytesToMB() => (this / 1024 / 1024).round();
```
Comment (bkonyi) [inline more_examples/autosnapshotting/lib/main.dart:15]: Nit: can this be `(this / (1024 * 1024)).round()` instead?

File: lib/src/usage_tracking/model.dart (hunk 3 above)
Comment (polina-c) [reply lib/src/usage_tracking/model.dart:1]: Yes, but the original was created in 2023.

File: lib/src/usage_tracking/_usage_event.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
```
[the remark this answers is not in the document]
Comment (polina-c) [reply lib/src/usage_tracking/_usage_event.dart:1]: Yes, but the original was created in 2023.

**Review by bkonyi — APPROVED** [review]

**Pull request merged**